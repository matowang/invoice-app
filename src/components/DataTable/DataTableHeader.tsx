import { TableCell } from "@mui/material";

import ArrowCircleUpOutlinedIcon from "@mui/icons-material/ArrowCircleUpOutlined";
import ArrowCircleDownOutlinedIcon from "@mui/icons-material/ArrowCircleDownOutlined";

import { DataTableField } from "./DataTable";
import { SortOrder } from "../../api/base";
import { MouseEventHandler } from "react";

export interface DataTableHeaderProps {
	onClick?: MouseEventHandler<HTMLTableCellElement>;
	field: DataTableField;
	index: number;
	sortOrder: SortOrder | undefined;
}

const DataTableHeader = ({ field, onClick, sortOrder, index }: DataTableHeaderProps) => (
	<TableCell
		onClick={onClick}
		align='right'
		component='th'
		className='font-bold cursor-pointer group'
		key={field.name}
		data-test={field["data-test"]}
	>
		<span className={`flex content-center relative justify-end group-first:justify-start`}>
			<div className='relative'>
				{sortOrder && (
					<div className='absolute left-0 -translate-x-full'>
						{sortOrder === "asc" ? (
							<ArrowCircleUpOutlinedIcon fontSize='small' />
						) : (
							<ArrowCircleDownOutlinedIcon fontSize='small' />
						)}
					</div>
				)}
				{field.label}
			</div>
		</span>
	</TableCell>
);

export default DataTableHeader;
