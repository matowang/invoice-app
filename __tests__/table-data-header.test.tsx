import { Table, TableHead, TableRow } from "@mui/material";
import { render } from "@testing-library/react";
import DataTableHeader from "../src/components/DataTable/DataTableHeaderField";

describe("table row message", () => {
	it("renders sortable properly", () => {
		const { container } = render(
			<Table>
				<TableHead>
					<TableRow>
						<DataTableHeader data-test='test'>Matthew</DataTableHeader>
					</TableRow>
				</TableHead>
			</Table>
		);
		expect(container).toMatchSnapshot();
	});
	it("renders non-sortable properly", () => {
		const { container } = render(
			<Table>
				<TableHead>
					<TableRow>
						<DataTableHeader sortOrder='asc' data-test='test'>
							Matthew
						</DataTableHeader>
					</TableRow>
				</TableHead>
			</Table>
		);
		expect(container).toMatchSnapshot();
	});
});
