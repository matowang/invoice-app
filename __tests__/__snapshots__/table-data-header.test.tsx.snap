// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table row message renders non-sortable properly 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium font-bold cursor-pointer group css-1azl6jz-MuiTableCell-root"
          data-test="test"
          scope="col"
        >
          <span
            class="flex content-center relative justify-end group-first:justify-start"
          >
            <div
              class="relative"
            >
              <div
                class="absolute left-0 -translate-x-full"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="ArrowCircleUpOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0 2c5.52 0 10-4.48 10-10S17.52 2 12 2 2 6.48 2 12s4.48 10 10 10zm-1-10v4h2v-4h3l-4-4-4 4h3z"
                  />
                </svg>
              </div>
              Matthew
            </div>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`table row message renders sortable properly 1`] = `
<div>
  <table
    class="MuiTable-root css-rqglhn-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium font-bold cursor-pointer group css-1azl6jz-MuiTableCell-root"
          data-test="test"
          scope="col"
        >
          <span
            class="flex content-center relative justify-end group-first:justify-start"
          >
            <div
              class="relative"
            >
              Matthew
            </div>
          </span>
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
